vault/expiration.go

```diff
@@ -332,12 +332,18 @@ func (m *ExpirationManager) RenewToken(req *logical.Request, source string, toke
 		return nil, err
 	}
 
-	// Fast-path if there is no renewal
 	if resp == nil {
 		return nil, nil
 	}
+
+	if resp.IsError() {
+		return &logical.Response{
+			Data: resp.Data,
+		}, nil
+	}
+
 	if resp.Auth == nil || !resp.Auth.LeaseEnabled() {
-		return resp.Auth, nil
+		return &logical.Response{}, nil
```

Comment:
In this case I'd go with the previous behavior -- set `Auth` in the new `logical.Response` to resp.Auth`. If it's `nil`, no harm no foul; if it's not, other information will continue through.